### PR TITLE
Support clone with relation (fix #381)

### LINKED ISSUE #381: Fail to clone object with relation
How to reproduce:

```
$ npm i -g parse
/path/to/lib
└─┬ parse@1.9.2 
  └─┬ babel-runtime@6.18.0 
    ├── core-js@2.4.1 
    └── regenerator-runtime@0.9.6 

$ node
> Parse = require('parse/node')
{ ... }
> o = Parse.Object.fromJSON({objectId: '7777777777', className: "Foo", aRelation: {__type: "Relation", className: "Foo"}})
ParseObject { _objCount: 0, className: 'Foo', id: '7777777777' }
> o2 = o.clone()
ParseObject { _objCount: 1, className: 'Foo' }
> o2.relation('aRelation')
Error: Internal Error. Relation retrieved from two different Objects.
    at ParseRelation.value (/path/to/node_modules/parse/lib/node/ParseRelation.js:70:19)
    at ParseObject.value (/path/to/node_modules/parse/lib/node/ParseObject.js:630:15)
    at repl:1:4
    at REPLServer.defaultEval (repl.js:272:27)
    at bound (domain.js:280:14)
    at REPLServer.runBound [as eval] (domain.js:293:12)
    at REPLServer.<anonymous> (repl.js:441:10)
    at emitOne (events.js:101:20)
    at REPLServer.emit (events.js:188:7)
    at REPLServer.Interface._onLine (readline.js:219:10)
```

I'll send PR later.

### PR DESCRIPTION
**Pull request by yuki-takeichi**
Fix #381

### CODE REVIEW

**Comment by yuki-takeichi** [comment]
signed.

**Comment by yuki-takeichi** [comment]
I fixed a bug related to this change. Sorry for confusing. It's mergeable now.

File: src/__tests__/ParseObject-test.js [hunk 1]
```diff
@@ -465,15 +456,46 @@ describe('ParseObject', () => {
       'Called relation() on non-relation field age'
     );
     var rel = o.relation('friends');
-    expect(rel.parentClass).toBe('Person');
-    expect(rel.parentId).toBe('AA');
```
Comment (flovilmart) [inline src/__tests__/ParseObject-test.js]: why this change? this seems to be a breaking on.

**Comment by flovilmart** [comment]
@yuki-takeichi ping

**Comment by dplewis** [comment]
Needed for https://github.com/parse-community/parse-dashboard/issues/924

@flovilmart I found a simpler solution and added more tests for this

**Comment by dplewis** [comment]
Its more of getting and saving relation on a cloned object.

I may have found a separate issue while doing this. I'll add a few more tests.

**Comment by dplewis** [comment]
How does this look?

**Comment by flovilmart** [comment]
Uhm. Ok, reading back the original issue I guess it makes sense.
